### PR TITLE
Add support for user assigned identities in Policy Assignments

### PR DESCRIPTION
**Pull request by calecarter**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. Already deployed
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. 7/19/2021
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/dataPolicyManifests.json [hunk 1]
```diff
@@ -0,0 +1,311 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "PolicyClient",
+    "version": "2021-06-01",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/dataPolicyManifests.json:5]: did you mean to pull this forward?  The manifest doesn't have this version for dataPolicyManifests right?  I'm not against it but check with dataplane and add to manifest if we do this

File: specification/resources/resource-manager/readme.md [hunk 2]
```diff
@@ -64,6 +64,23 @@ tag: package-deploymentscripts-2020-10
 tag: package-templatespecs-2021-03-preview
 ```
 
+### Tag: package-policy-2020-09
```
Comment (pilor) [inline specification/resources/resource-manager/readme.md:67]: 2021-06

**Review by pilor — CHANGES_REQUESTED** [review]

**Comment by pilor** [comment]
Some of the other language specific readme files need to be updated with the new tag.  Search for "package-policy-2020-09" to see where that was being used

**Comment by raych1** [comment]
@calecarter , it has cross version breaking change against last preview version and needs to go thru breaking change review. Also please fix prettierCheck failures.

**Comment by raych1** [comment]
@calecarter , please also fix other CI check failures.

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/dataPolicyManifests.json (hunk 1 above)
Comment (calecarter) [reply specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/dataPolicyManifests.json]: I did initially, but chatted with Varun and he didn't want me to do it.  So I reverted it.

File: specification/resources/resource-manager/readme.md (hunk 2 above)
Comment (calecarter) [reply specification/resources/resource-manager/readme.md]: Thanks...  Fixed this.

**Comment by calecarter** [comment]
Updated those files too.  Thx.

---
In reply to: [879409400](https://github.com/Azure/azure-rest-api-specs/pull/15181#issuecomment-879409400) [](http://example.com/codeflow?ancestors=879409400,879409400)

**Comment by pilor** [comment]
Holding ARMSignOff until PATCH + SystemData is in.  Otherwise, the next PR that introduces them in this api-version will be seen as a breaking change by the tooling

**Comment by calecarter** [comment]
> @calecarter , it has cross version breaking change against last preview version and needs to go thru breaking change review. Also please fix prettierCheck failures.

There should not be any breaking changes.  These are just additive changes.


> Holding ARMSignOff until PATCH + SystemData is in. Otherwise, the next PR that introduces them in this api-version will be seen as a breaking change by the tooling

updated with PATCH + SystemData

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/policyAssignments.json [hunk 3]
```diff
@@ -0,0 +1,980 @@
...
+      },
+      "patch": {
+        "tags": [
+          "PolicyAssignments"
+        ],
+        "operationId": "PolicyAssignments_Update",
+        "summary": "Updates a policy assignment.",
+        "description": " This operation updates a policy assignment with the given scope and name. Policy assignments apply to all resources contained within their scope. For example, when you assign a policy at resource group scope, that policy applies to all resources in the group.",
+        "x-ms-examples": {
+          "Create or update a policy assignment with a system assigned identity": {
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/policyAssignments.json:217]: Examples for patch should just say "Update"

**Review by pilor — APPROVED** [review]

File: specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/policyAssignments.json (hunk 3 above)
Comment (calecarter) [reply specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/policyAssignments.json]: fixed...  thx

**Comment by raych1** [comment]
@msyyc @ArcturusZhang , can you confirm the breaking change for SDK?

**Comment by msyyc** [comment]
approved for python

**Comment by raych1** [comment]
@ArcturusZhang , can you confirm the breaking change for GO SDK?

**Comment by calecarter** [comment]
@raych1 anything I need to do here or we just waiting on Go SDK confirmation?

**Pull request merged**